src/architecture/Module.h

```diff
@@ -82,6 +87,39 @@ class Module {
      * @returns What type the channel has.
      */
     ModuleType getModuleType() const;
+
+    virtual void lbmInit(T dynViscosity, T density) {}
```

Comment:
From here, the functions are not relevant to the architecture, but manipulate solver parameters. These should go in a `simulation/CFDSolver.h` header file. Same goes for lbmModule and essLbmModule, which should be renamed and moved to the `simulation` folder.